box-content-sdk/src/test/java/com/box/androidsdk/content/requests/BoxUploadRequestTest.java

```diff
@@ -0,0 +1,116 @@
+        String sampleResponseJson = "{ \"total_count\": 1, \"entries\": [ { \"type\": \"file\", \"id\": \"5000948880\", \"sequence_id\": \"3\", \"etag\": \"3\", \"sha1\": \"134b65991ed521fcfe4724b7d814ab8ded5185dc\", \"name\": \"tigers.jpeg\", \"description\": \"a picture of tigers\", \"size\": 629644, \"path_collection\": { \"total_count\": 2, \"entries\": [ { \"type\": \"folder\", \"id\": \"0\", \"sequence_id\": null, \"etag\": null, \"name\": \"All Files\" }, { \"type\": \"folder\", \"id\": \"11446498\", \"sequence_id\": \"1\", \"etag\": \"1\", \"name\": \"Pictures\" } ] }, \"created_at\": \"2012-12-12T10:55:30-08:00\", \"modified_at\": \"2012-12-12T11:04:26-08:00\", \"trashed_at\": null, \"purged_at\": null, \"content_created_at\": \"2013-02-04T16:57:52-08:00\", \"content_modified_at\": \"2013-02-04T16:57:52-08:00\", \"created_by\": { \"type\": \"user\", \"id\": \"17738362\", \"name\": \"sean rose\", \"login\": \"sean@box.com\" }, \"modified_by\": { \"type\": \"user\", \"id\": \"17738362\", \"name\": \"sean rose\", \"login\": \"sean@box.com\" }, \"owned_by\": { \"type\": \"user\", \"id\": \"17738362\", \"name\": \"sean rose\", \"login\": \"sean@box.com\" }, \"shared_link\": null, \"parent\": { \"type\": \"folder\", \"id\": \"11446498\", \"sequence_id\": \"1\", \"etag\": \"1\", \"name\": \"Pictures\" }, \"item_status\": \"active\" } ] }";
+
+
+
+        //Byte stream to capture the outputstream to the server
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(1024);
+
+        //Mock httpurlconnection for uploads
+        URL u = PowerMockito.mock(URL.class);
+        PowerMockito.whenNew(URL.class).withAnyArguments().thenReturn(u);
+        HttpURLConnection huc = PowerMockito.mock(HttpURLConnection.class);
+        PowerMockito.when(u.openConnection()).thenReturn(huc);
+        PowerMockito.when(huc.getOutputStream()).thenReturn(outputStream);
+        PowerMockito.when(huc.getContentType()).thenReturn(BoxRequest.ContentTypes.JSON.toString());
+        PowerMockito.when(huc.getResponseCode()).thenReturn(200);
+
+        //Mock Response to use sample json
+        BoxHttpResponse foo = new BoxHttpResponse(huc);
+        PowerMockito.when(foo.getBody()).thenReturn(new ByteArrayInputStream(sampleResponseJson.getBytes()));
+        PowerMockito.whenNew(BoxHttpResponse.class).withAnyArguments().thenReturn(foo);
+
+
+        BoxApiFile fileApi = new BoxApiFile(newMockBoxSession());
+        InputStream inputStream = new ByteArrayInputStream(requestUrl.getBytes());
+        BoxRequestsFile.UploadFile uploadRequest = fileApi.getUploadRequest(inputStream, "filename", "0");
+        Assert.assertEquals("filename", uploadRequest.getFileName());
```

Comment:
Getter are tested in the test below do we need those assert here? Tests should focus on one thing at the time.